src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -259,7 +263,31 @@ private static string FindBestVersionMatch(VersionRange versionRange, IEnumerabl
                 return version;
             }).Where(v => v != null);
 
-            return bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault()?.ToString();
+            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+
+            if (bundleId != ScriptConstants.DefaultExtensionBundleId)
+            {
```

Comment:
Trying to understand which scenario will execute this `if-block`? Can you share examples of possible values of `bundleId` here?

Reply:
We only want to match and limit the bundle version for the official bundle version
`Microsoft.Azure.Functions.ExtensionBundle`

Other possible values are.
`Microsoft.Azure.Functions.ExtensionBundle.Preview`
`Microsoft.Azure.Functions.ExtensionBundle.Workflows`